### DISCUSSION TITLE
MultiCore RP Pico main code on core0 being controlled by TFT buttons running on core1

### DISCUSSION THREAD

**Discussion by BrianB03** [earlephilhower/arduino-pico#2280] [General]
Hi Earl, I need to pass vars belonging to 'buttons' on a TFT touch screen (monitored with code running on core1/loop1() to
my main program running within core0/loop().
N.B. The main program runs in loops of 2.5 seconds for approx 13 minutes then stops. I would like a button press to be sent
over immediately (like an interrupt)
Please advise an approach .. many thanks Brian

**Comment by earlephilhower** [comment]
Global variables are an easy way to do this.  Make sure they're `volatile` and 32-bits in size so that they won't be cached or face read-modify-write race conditions.  W/o knowing more specifics, that would be an easy suggestion. to implement.
